packages/monaco/src/browser/monaco-command.ts

```diff
@@ -120,6 +127,93 @@ export class MonacoEditorCommandHandlers implements CommandContribution {
             }
         };
     }
+    protected newConfigIndentationHandler(): MonacoEditorCommandHandler {
+        return {
+            execute: (editor: MonacoEditor) => {
+                const configTabSize = (useSpaces: boolean) => {
+                    const editorModel = editor.document;
+                    if (editorModel && editorModel.textEditorModel) {
+                        const tabSize = editorModel.textEditorModel.getOptions().tabSize;
+                        const configuredTabSize = this.editorPreferences['editor.tabSize'];
+                        const sizes = Array.from(Array(8), (_, x) => x + 1);
+                        const tabSizeOptions = sizes.map(size =>
+                            new UpdateTabSizeOpenItem(size === configuredTabSize ? `${size}   Configured Tab Size` : size.toString(),
+                                useSpaces,
+                                () => {
+                                    editorModel.textEditorModel.updateOptions({
+                                        tabSize: size || tabSize,
+                                        insertSpaces: useSpaces
+                                    });
+                                },
+                                (mode: QuickOpenMode) => mode === QuickOpenMode.OPEN,
+                                () => true
+                            ));
+                        this.open(tabSizeOptions,
+                            'Select Tab Size for Current File',
+                            (lookFor: string) => {
+                                if (!lookFor || lookFor === '') {
+                                    return tabSize - 1;
+                                }
+                                return sizes.findIndex(s => lookFor.startsWith(s.toString()));
+                            });
```

Comment:
Weird thing is that, the auto-highlight does not always work. I am not sure if it is a bug in my code or a problem in the QuickOpen. Please see my demo:
![peek 2018-03-26 02-41](https://user-images.githubusercontent.com/37082801/37890654-5332f6b0-309f-11e8-94ae-9ac6986a9f12.gif)

Reply:
Could you debug and find a root cause?

Reply:
I tested a few scenarios where the highlighting works and doesn't work. All cases ended up sending the same data into `QuickOpenWidget.setInput()` function:
https://github.com/theia-ide/theia/blob/master/packages/monaco/src/typings/monaco/index.d.ts#L365

Looks like it is a problem outside theia.

Reply:
ok, could you open a follow up issue?

Reply:
Like I said it looks like a problem outside theia, so ... where am I supposed to open that ticket ?